src/Microsoft.QualityGuidelines.Analyzers/Core/UseLiteralsWhereAppropriate.Fixer.cs

```diff
@@ -22,9 +28,82 @@ public sealed override FixAllProvider GetFixAllProvider()
 
         public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
         {
-            // This is to get rid of warning CS1998, please remove when implementing this analyzer
-            await new Task(() => { }).ConfigureAwait(false);
-            throw new NotImplementedException();
+            SyntaxNode root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            SyntaxNode declaration = root.FindNode(context.Span);
+            declaration = SyntaxGenerator.GetGenerator(context.Document).GetDeclaration(declaration, DeclarationKind.Field);
+            var fieldFeclaration = GetFieldDeclaration(declaration);
+            if (fieldFeclaration == null)
+            {
+                return;
+            }
+
+            // We cannot have multiple overlapping diagnostics of this id.
+            Diagnostic diagnostic = context.Diagnostics.Single();
+
+            context.RegisterCodeFix(
+                new MyCodeAction(
+                    MicrosoftQualityGuidelinesAnalyzersResources.UseLiteralsWhereAppropriateCodeActionTitle,
+                    cancellationToken => ToConstantDeclarationAsync(context.Document, fieldFeclaration, cancellationToken)),
+                diagnostic);
+        }
+
+        private async Task<Document> ToConstantDeclarationAsync(Document document, SyntaxNode fieldDeclaration, CancellationToken cancellationToken)
```

Comment:
Thanks for the suggestion.

I tried to replace the body of this method with:
```csharp
            var syntaxRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
            var editor = new SyntaxEditor(syntaxRoot, document.Project.Solution.Workspace);
            editor.SetModifiers(fieldDeclaration, editor.Generator.GetModifiers(fieldDeclaration) - DeclarationModifiers.Static - DeclarationModifiers.ReadOnly + DeclarationModifiers.Const);
            var newSyntaxRoot = editor.GetChangedRoot();
            return document.WithSyntaxRoot(newSyntaxRoot);
```

But that breaks three tests. In-between comments are not preserved.

For example:
```
Message: Assert.Equal() Failure
                                 ↓ (pos 43)
Expected: ···c /*leading*/ const /*intermediate*/  /*trailing*/ string f1 ···
Actual:   ···c /*leading*/ const string f1 = "";\r\n}\r\n
                                 ↑ (pos 43)
```